File: LeetCodeSolutions/LeetCode_0014.py
from typing import List


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if strs == []:
            return ''
        n = min([len(_) for _ in strs])
        if n < 1:
            return ''
        res = ''
        i = 0
        while i < n:
            s = [_[i] for _ in strs]
            tmp = s[0]
            s = set(s)
            if len(s) == 1:
                res += tmp
            else:
                break
            i += 1
        return res

File: LeetCodeSolutions/LeetCode_0014.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int n = strs.size();
        if (n == 0) return {};
        string ret = strs[0];
        for (int i = 1; i < n; ++i){
            ret = lcs(ret, strs[i]);
        }
        return ret;

    }

    string lcs(string str1, string str2){
        if (str1 == "" || str2  == "") return {};
        string ret;
        for(int i = 0; i < str1.size() && i < str2.size(); ++i){
            if (str1[i] == str2[i]) ret += str1[i];
            else break;
        }
        return ret;
    }
};